src/main/java/servlets/SearchServlet.java

```diff
@@ -62,24 +65,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     // Convert matching receipt entities to receipt objects and add to return list.
-    Spliterator<Entity> spliterator = datastore.prepare(query).asIterable().spliterator();
-    Stream<Entity> entities = StreamSupport.stream(spliterator, false);
-    return entities
-        .map(entity -> {
-          long id = entity.getKey().getId();
-          long userId = (long) entity.getProperty("userId");
-          long timestamp = (long) entity.getProperty("timestamp");
-          BlobKey blobKey = (BlobKey) entity.getProperty("blobKey");
-          String imageUrl = (String) entity.getProperty("imageUrl");
-          double price = (double) entity.getProperty("price");
-          String store = (String) entity.getProperty("store");
-          String label = (String) entity.getProperty("label");
-          Set<String> categories =
-              new HashSet<String>((ArrayList) entity.getProperty("categories"));
-          String rawText = (String) entity.getProperty("rawText");
-          return new Receipt(
-              id, userId, timestamp, blobKey, imageUrl, price, store, label, categories, rawText);
-        })
-        .collect(Collectors.toList());
+    return ImmutableList.copyOf(datastore.prepare(query)
+                                    .asList(FetchOptions.Builder.withDefaults())
+                                    .stream()
+                                    .map(entity -> { return createReceiptFromEntity(entity); })
+                                    .collect(Collectors.toList()));
+  }
+
+  /** Create and return a java Receipt object from a datastore Receipt entity. */
```

Comment:
You can use `{@link Receipt}` to directly reference the Receipt object in javadocs. Same could be done with Entity.